inspirehep/ext/cache_manager/cache.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import functools
+
+import logging
+
+from flask import current_app, request
+
+from flask_cache import Cache
+
+from invenio_base.globals import cfg
+
+logger = logging.getLogger(__name__)
+
+
+class InspireCache(Cache):
+    """Extends Cache from flask_cache and adds mode-checking (debug or not) for memoize and cached methods."""
+
+    def inspire_memoize(self, timeout=None, make_name=None, unless=None):
```

Comment:
Why are you calling it `inspire_memoize`, rather than overriding the parent class?